metricflow/model/transformations/dbt_to_metricflow.py

```diff
@@ -264,51 +264,44 @@ def merge_data_sources(cls, data_sources: List[DataSource]) -> DataSource:
         measures: Set[Measure] = set()
         identifiers: Set[Identifier] = set()
         dimensions: Set[Dimension] = set()
-        names: List[str] = []
-        descriptions: List[str] = []
-        sql_tables: List[str] = []
-        sql_queries: List[str] = []
-        dbt_models: List[str] = []
+        names: Set[str] = set()
+        descriptions: Set[str] = set()
+        sql_tables: Set[str] = set()
+        sql_queries: Set[str] = set()
+        dbt_models: Set[str] = set()
         for data_source in data_sources:
-            names.append(data_source.name) if data_source.name else None
-            descriptions.append(data_source.description) if data_source.description else None
-            sql_tables.append(data_source.sql_table) if data_source.sql_table else None
-            sql_queries.append(data_source.sql_query) if data_source.sql_query else None
-            dbt_models.append(data_source.dbt_model) if data_source.dbt_model else None
+            names.add(data_source.name) if data_source.name else None
+            descriptions.add(data_source.description) if data_source.description else None
+            sql_tables.add(data_source.sql_table) if data_source.sql_table else None
+            sql_queries.add(data_source.sql_query) if data_source.sql_query else None
+            dbt_models.add(data_source.dbt_model) if data_source.dbt_model else None
```

Comment:
Oh I just noticed this is odd, because it's updating in place and that `None` gets dropped on the floor. I guess the only other options are if statements or repeating the iteration for a set of comprehensions like:

```
names = {data_source.name for data_source in data_sources if data_source.name}
descriptions = {data_source.description for data source in data_sources if data_source.description}
# ....
```

More readable, in my opinion, but it does more work. Your call!

Reply:
I think I'm willing to take the hit on readability here for doing less work. I think we're gonna run up against large dbt metric configs relatively quickly. I'll add a comment detailing what's going on here 😅